[pytorch] Add logger for pt2 compile chromium events to hive

ppanchalia opened the pull request:
Summary:
X-link: https://github.com/pytorch/benchmark/pull/2535

Logging raw chromium events to hive per job run enables us to build combined rank perfetto traces without having to depend on Logarithm and deal with things like rate limits etc.

We can easily build a utility to query hive and upload traces to manifold and view them on perfetto

Test Plan:
Launch a job

```
buck2 run mode/opt //aps_models/examples/dlrm:dlrm_train_app -- --config-name train_mast_fsdp_torchdynamo launcher.data_project=apf_ai_infra launcher.fbl_entitlement=ai_infra_training_rnd_tc  launcher.hardware=TC_ANY_80G
```

Local run
```
Perfetto: ['https://interncache-all.fbcdn.net/manifold/perfetto-artifacts/tree/ui/index.html?url=https://interncache-all.fbcdn.net/manifold/pt2_compile_traces_test/tree/pt2_trace_files/aps-ppanchalia-426838c277/0/0/2bc9975d-921c-4766-9cb2-e7ce9833ae96.json']
```

{F1954710538}

Differential Revision: D65525513




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

ppanchalia commented:
@pytorchbot label "topic: not user facing"

jamesjwu commented on torch/_dynamo/utils.py:
```diff
@@ -1175,6 +1176,7 @@ def _log_timed_event(
             suppress_context=False,
             expect_trace_id=False,  # Not every chromium event will have a trace_id
         )
+        log_pt2_compile_chromium_event(event)
```
FWIW, reading this from the internal diff makes sense, but reading it externally I see now that this naming is pretty impossible to understand (when compared to log_pt2_compile_event). 

We have:

- ChromiumEventLogger, which logs a chromium event to tlparse 
- log_pt2_compile_event, which logs to PT2 Compile Events table internally
- And this, `log_pt2_compile_chromium_event`, which logs to a *different* table internally. 

Can we call this something different, like `log_internal_chromium_event` or `record_chromium_event_internal`? 

It would distinguish this table from the Pt2 Compile Events table.

ppanchalia replied:
done

jamesjwu reviewed (approved):
Thanks so much!

ppanchalia committed:
[pytorch] Add logger for pt2 compile chromium events to hive (#139941)

Summary:

X-link: https://github.com/pytorch/benchmark/pull/2535

Logging raw chromium events to hive per job run enables us to build combined rank perfetto traces without having to depend on Logarithm and deal with things like rate limits etc.

We can easily build a utility to query hive and upload traces to manifold and view them on perfetto

Test Plan:
Launch a job

```
buck2 run mode/opt //aps_models/examples/dlrm:dlrm_train_app -- --config-name train_mast_fsdp_torchdynamo launcher.data_project=apf_ai_infra launcher.fbl_entitlement=ai_infra_training_rnd_tc  launcher.hardware=TC_ANY_80G
```

Local run
```
Perfetto: ['https://interncache-all.fbcdn.net/manifold/perfetto-artifacts/tree/ui/index.html?url=https://interncache-all.fbcdn.net/manifold/pt2_compile_traces_test/tree/pt2_trace_files/aps-ppanchalia-426838c277/0/0/2bc9975d-921c-4766-9cb2-e7ce9833ae96.json']
```

{F1954710538}

Reviewed By: jamesjwu

Differential Revision: D65525513

ppanchalia commented:
@pytorchbot merge
